admin/class-metabox.php

```diff
@@ -182,69 +182,21 @@ public function add_meta_box() {
 	}
 
 	/**
-	 * Pass some variables to js for the edit / post page overview, snippet preview, etc.
+	 * Pass variables to js for use with the post-scraper
 	 *
-	 * @return  array
+	 * @return array
 	 */
-	public function localize_script() {
-		$post = $this->get_metabox_post();
-
-		if ( ( ! is_object( $post ) || ! isset( $post->post_type ) ) || $this->is_metabox_hidden( $post->post_type ) === true ) {
-			return array();
-		}
-
-		$options = get_option( 'wpseo_titles' );
-
-		$date = '';
-		if ( isset( $options[ 'showdate-' . $post->post_type ] ) && $options[ 'showdate-' . $post->post_type ] === true ) {
-			$date = $this->get_post_date( $post );
-
-			self::$meta_length        = ( self::$meta_length - ( strlen( $date ) + 5 ) );
-			self::$meta_length_reason = __( ' (because of date display)', 'wordpress-seo' );
-		}
-
-		self::$meta_length_reason = apply_filters( 'wpseo_metadesc_length_reason', self::$meta_length_reason, $post );
-		self::$meta_length        = apply_filters( 'wpseo_metadesc_length', self::$meta_length, $post );
-
-		unset( $date );
-
-		$title_template = '';
-		if ( isset( $options[ 'title-' . $post->post_type ] ) && $options[ 'title-' . $post->post_type ] !== '' ) {
-			$title_template = $options[ 'title-' . $post->post_type ];
-		}
-
-		// If there's no title template set, use the default, otherwise title preview won't work.
-		if ( $title_template == '' ) {
-			$title_template = '%%title%% - %%sitename%%';
-		}
-
-		$metadesc_template = '';
-		if ( isset( $options[ 'metadesc-' . $post->post_type ] ) && $options[ 'metadesc-' . $post->post_type ] !== '' ) {
-			$metadesc_template = $options[ 'metadesc-' . $post->post_type ];
-		}
-
-		$sample_permalink = get_sample_permalink( $post->ID );
-		$sample_permalink = str_replace( '%page', '%post', $sample_permalink[0] );
+	public function localize_post_scraper_script() {
+		$json = $this->get_scraper_json();
```

Comment:
If this returns translations we should rename the function to `get_scraper_translations`